atc/atccmd/config_types.go

```diff
@@ -0,0 +1,208 @@
+package atccmd
+
+import (
+	"net"
+	"time"
+
+	"github.com/concourse/concourse/atc/creds"
+	"github.com/concourse/concourse/atc/creds/conjur"
+	"github.com/concourse/concourse/atc/creds/credhub"
+	"github.com/concourse/concourse/atc/creds/dummy"
+	"github.com/concourse/concourse/atc/creds/kubernetes"
+	"github.com/concourse/concourse/atc/creds/secretsmanager"
+	"github.com/concourse/concourse/atc/creds/ssm"
+	"github.com/concourse/concourse/atc/creds/vault"
+	"github.com/concourse/concourse/atc/metric"
+	"github.com/concourse/concourse/atc/metric/emitter"
+	"github.com/concourse/concourse/atc/policy"
+	"github.com/concourse/concourse/atc/worker"
+	"github.com/concourse/concourse/skymarshal/skycmd"
+	"github.com/concourse/flag"
+)
+
+type TLSConfig struct {
+	BindPort uint16    `yaml:"bind_port,omitempty"`
+	Cert     flag.File `yaml:"cert,omitempty"`
+	Key      flag.File `yaml:"key,omitempty"`
+	CaCert   flag.File `yaml:"ca_cert,omitempty"`
```

Comment:
Could make sense to support inlining these values rather than referencing files, but referencing files makes sense here too.

Maybe these could be called `*_file` and the inline form could be named `cert`, `ca_cert`, `key`?